### ISSUE TITLE
Need some detailed info on the execution

### ISSUE THREAD

**Issue by kunwar-vikrant** [ShengranHu/ADAS#12]
Hello all,
I need some help understanding the flow. How do you guys actually use the results, how is it modified to perform some task, how do you adapt it to use custom agents. I would be grateful if someone could explain it.

**Comment by hemangjoshi37a** [comment]
same here. better if someone can provide any documentation for this

**Comment by Bilgeworth** [comment]
Best guess so far, text formatted by a bot:

Each folder starting with an underscore (`_`) is an LLM benchmark. The `search.py` script within each benchmark folder defines the setup, execution, and evaluation process for that benchmark.

The `search.py` script is responsible for creating agents, testing them, and iterating to optimize their performance. The evaluation of these agents is handled by the `evaluate_forward_fn` function within the script.

To run a benchmark, navigate to the repository's root directory and execute the `search.py` script for the desired benchmark. For example, to run the ARC benchmark, use the following command:

```bash
python _arc/search.py
```

This will create agents, test them, iterate to improve their performance, and output a JSON file with the results.

If you want to create your own benchmark, follow these steps:

Copy an Existing Benchmark: Choose the most similar existing benchmark folder, copy it, and rename it to your desired benchmark name.
Modify the Benchmark: Swap in your benchmark-specific setup, execution, and evaluation logic in the search.py script.
Run the Benchmark: Execute the search.py script for your new benchmark. Ensure you change the output name to avoid overwriting the results of the default benchmarks.

Evaluation Function
The actual testing of agents is performed by the evaluate_forward_fn function in search.py. 
For example, in the _arc benchmark, this function includes a link to the GitHub repository where the evaluation criteria were sourced.

**Comment by kunwar-vikrant** [comment]
Thank you for the information!, have you been able to deploy the agents for any custom tasks?

**Comment by Bilgeworth** [comment]
Haven't had any luck yet, biggest issue I'm running into is defining a scoring system, in the mean time I've been doing my best to make something easier to edit on a fork, here's the flowchart from that, some variable and function names have been changed but the workflow is the same (some issues like calculating fitness twice without making changes between have been removed)
![image](https://github.com/user-attachments/assets/d158ecbe-c78a-4b89-b99d-853ae1c12270)

**Comment by Bilgeworth** [comment]
Realized the previous is a bit hard to follow, here's it color coded and rearranged
![image](https://github.com/user-attachments/assets/48b97054-388a-4f84-9039-9e4dde372d88)

**Comment by wofeichangaiwoai** [comment]
I have tried to transfer this to other custom domains. my domain use the agent to design a code system, it is not very easy to evaluate the final result of this agent. so how can I use adas if our domain don't have any true label? From my understanding, the adas will evaluate the dataset first, and then enhance the final result via the whole pipeline, right?

**Comment by ShengranHu** [comment]
@Bilgeworth Thank you very much for your detailed explanation and for the nice flowchart! 

@wofeichangaiwoai We believe developing ADAS algorithms for domains that don't have ground truth labels will be an important future research direction. Currently, we anticipate that an LLM evaluator might be used to provide feedback signals for the mate agent.
